networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/api/P2PNetwork.java

```diff
@@ -24,6 +28,18 @@
    */
   CompletableFuture<?> connect(String peer);
 
+  Optional<? extends Peer> getPeer(NodeId id);
+
+  Stream<? extends Peer> streamPeers();
+
+  default long getPeerCount() {
+    return streamPeers().count();
```

Comment:
This is a pretty expensive way to get the peer count as it has to iterate over all the peers each time it's called. It's probably better to keep getting this from `PeerManager` so it can use `connectedPeerMap.size()`. Admittedly it's unlikely to be a problem in practice, but having methods that look cheap and turn out to take linear time risks them winding up being used inside a loop and suddenly you have exponential time and things get messy.

Reply:
I removed the default implementation.  But I ended up just leaving a TODO on this for the `Eth2Network` implementation: https://github.com/PegaSysEng/artemis/pull/1042/commits/0fc62a8f30cc4f2d32bb04baf4a8858873c16698#diff-24b16da25038d7f0ac1d55d0f2ba6daeR42-R47